File: backend/src/main/java/com/seshutechie/finociate/model/ReportItemData.java
package com.seshutechie.finociate.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;

import java.util.List;
import java.util.Map;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ReportItemData {
    private String title;
    private List<String> groupByFields;
    private String aggregationType; // Overall, Monthly
    private String displayType; // Table, Chart
    private String valueType; // Table, Chart
    private Map<String, List<ReportItemEntry>> itemEntries;
}

File: finociate-ui/src/app/shared/model/report-item-data.ts
import { ReportItemEntry } from "./report-item-entry";

interface EntryMap {
    [key: string]: ReportItemEntry[];
}

export interface ReportItemData {
    title: string,
    groupByFields: string[],
    displayType: string,
    aggregationType: string,
    valueType: string,
    itemEntries: EntryMap,
}
